pkg/certificate/identity.go

```diff
@@ -401,14 +401,19 @@ func PrepareRoleCsrOptions(idCfg *config.IdentityConfig, domain, service string)
 
 	var roleCsrOptions []util.CSROptions
 
-	if len(idCfg.TargetDomainRoles) == 0 || idCfg.RoleCertDir == "" {
-		log.Debugf("Skipping to prepare csr for role certificates with target roles[%s], output directory[%s]", idCfg.TargetDomainRoles, idCfg.RoleCertDir)
+	if len(idCfg.RoleCertTargetDomainRoles) == 0 || idCfg.RoleCertDir == "" {
+		log.Debugf("Skipping to prepare csr for role certificates with target roles[%s], output directory[%s]", idCfg.RoleCertTargetDomainRoles, idCfg.RoleCertDir)
 		return nil, nil
 	}
 
-	for _, dr := range idCfg.TargetDomainRoles {
+	for _, dr := range idCfg.RoleCertTargetDomainRoles {
 		targetDomain, targetRole := dr.Domain, dr.Role
 
+		// Rolecert requires a role name in the CN, so it will skip if there is no role name.
+		if dr.Role == "" {
+			continue
+		}
```

Comment:
no need?

Reply:
fixed in: https://github.com/AthenZ/k8s-athenz-sia/pull/136/commits/10af4d896c5e1c7ba92fc554bf36ac27b7d54f29